.github/workflows/maven-release.yml

```diff
@@ -0,0 +1,49 @@
+name: Release to Maven Staging
+
+on:
+  push:
+    branches:
+      - master
+  # temporary to test it's working
+  pull_request:
+
+jobs:
+  build:
+    name: Build project and publish release
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK
+      uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 8
+        cache: maven
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+        gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+        gpg-passphrase: MAVEN_GPG_PASSPHRASE
+    - name: Extract project version
+      id: project
+      run: echo ::set-output name=version::$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
```

Comment:
We could extract the version into a project.properties, would probably make the version updates cleaner.

Reply:
I'd still use the help, builder-helper and the versions plugins though. Maybe I just don't picture how it would change in terms of the workflow.

Reply:
I'd resolve it for now to merge this one, but if you have an idea for improvement we can do it as part of this one: https://github.com/acm19/aws-request-signing-apache-interceptor/issues/53 or in a new issue.